src/torchmetrics/functional/detection/map.py

```diff
@@ -0,0 +1,315 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import contextlib
+import io
+from typing import Any, Dict, List, Literal, Optional, Tuple, Union
+
+import numpy as np
+import torch
+from lightning_utilities import apply_to_collection
+from torch import Tensor
+
+from torchmetrics.detection.helpers import CocoBackend, _get_safe_item_values, _input_validator, _validate_iou_type_arg
+from torchmetrics.utilities.imports import (
+    _FASTER_COCO_EVAL_AVAILABLE,
+    _PYCOCOTOOLS_AVAILABLE,
+    _TORCHVISION_AVAILABLE,
+)
+
+if not (_PYCOCOTOOLS_AVAILABLE or _FASTER_COCO_EVAL_AVAILABLE) or not _TORCHVISION_AVAILABLE:
+    __doctest_skip__ = [
+        "mean_average_precision",
+    ]
+
+
+def mean_average_precision(
+    preds: List[Dict[str, Any]],
+    target: List[Dict[str, Any]],
+    box_format: Literal["xyxy", "xywh", "cxcywh"] = "xyxy",
+    iou_type: Union[Literal["bbox", "segm"], Tuple[Literal["bbox", "segm"], ...]] = "bbox",
+    iou_thresholds: Optional[list[float]] = None,
+    rec_thresholds: Optional[list[float]] = None,
+    max_detection_thresholds: Optional[list[int]] = None,
+    class_metrics: bool = False,
+    extended_summary: bool = False,
+    average: Literal["macro", "micro"] = "macro",
+    backend: Literal["pycocotools", "faster_coco_eval"] = "pycocotools",
+    warn_on_many_detections: bool = True,
+) -> Union[Tensor, Dict[str, Tensor]]:
+    r"""Compute the mean average precision (mAP) and mean average recall (mAR) for object detection predictions.
+
+    This function evaluates detection predictions for either bounding boxes or segmentation masks based
+    on the provided ``iou_type``, comparing predictions (``preds``) and ground truth annotations (``target``)
+    using a COCO-style evaluation. The expected input for each image is a dictionary with keys:
+
+    - For bounding boxes (``iou_type="bbox"``): ``boxes``, ``scores``, and ``labels``.
+    - For segmentation (``iou_type="segm"``): ``masks``, ``scores``, and ``labels``.
+
+    In addition, ground truth dictionaries may include the optional keys ``iscrowd`` and ``area``.
+    Boxes are expected in the coordinate format provided via ``box_format``, which supports:
+    - ``"xyxy"``: [xmin, ymin, xmax, ymax]
```

Comment:
```suggestion
    Boxes are expected in the coordinate format provided via ``box_format``, which supports:

    - ``"xyxy"``: [xmin, ymin, xmax, ymax]
```
a single white space here i think is needed

Reply:
fixed